uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/APIWrapper/DefaultLimitRemoverTest.php

```diff
@@ -21,43 +29,37 @@ public function setUpHeadless() {
   }
 
   public function setUp() {
-    $this->defaultLimitRemoverObj = new CRM_HRCore_APIWrapper_DefaultLimitRemover();
-
-    $this->testParameters['params'] = [
-      'action' => 'get',
-      'entity' => 'contact',
-      'id' => 1,
-    ];
+    $this->defaultLimitRemover = new DefaultLimitRemover();
   }
 
-  public function testFromApiInputWithLimitSetInParameters() {
+  public function testTheDefaultLimitShouldNotBeAppliedIfASpecificLimitHasBeenSet() {
     $this->testParameters['params']['options']['limit'] = 10;
 
-    $actualValue =  $this->defaultLimitRemoverObj->fromApiInput($this->testParameters);
+    $actualValue =  $this->defaultLimitRemover->fromApiInput($this->testParameters);
 
     $this->assertEquals(
       $this->testParameters['params']['options']['limit'],
       $actualValue['params']['options']['limit']
     );
   }
 
-  public function testFromApiInputWithLimitNotSetInParameters() {
-    $actualValue =  $this->defaultLimitRemoverObj->fromApiInput($this->testParameters);
+  public function testTheDefaultNoLimitValueShouldBeAppliedIfNoLimitHasBeenSet() {
+    $actualValue =  $this->defaultLimitRemover->fromApiInput($this->testParameters);
```

Comment:
So here is the problem I'm talking about. There's no way you can be sure of what `$this->testParameters` will look like. It might have been changed by another test and will make this test fail